From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Andy Chiang <AndyChiang_git@outlook.com>
Date: Tue, 28 Oct 2025 14:34:29 +0700
Subject: [PATCH] fortify-headers: fix __fh_overlap() build error

This fixes the following compile error when compiling dtc:
```
../../../../staging_dir/toolchain-x86_64_gcc-15.2.0_musl/include/fortify/string.h: In function 'memcpy':
../../../../staging_dir/toolchain-x86_64_gcc-15.2.0_musl/include/fortify/fortify-headers.h:136:31: error: cast discards 'const' qualifier from pointer target type [-Werror=cast-qual]
```

Signed-off-by: Andy Chiang <AndyChiang_git@outlook.com>
---
 include/fortify-headers.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/include/fortify-headers.h
+++ b/include/fortify-headers.h
@@ -133,8 +133,8 @@
  * since gcc seems to like to generate code that relies on dst == src */
 #define __fh_overlap(a, len_a, b, len_b) \
 	( \
-		((char*)(a) < (char*)(b) && (char*)(b) < ((char*)(a) + (__fh_size_t)(len_a))) \
-	     || ((char*)(b) < (char*)(a) && (char*)(a) < ((char*)(b) + (__fh_size_t)(len_b))) \
+		((const char*)(a) < (const char*)(b) && (const char*)(b) < ((const char*)(a) + (__fh_size_t)(len_a))) \
+	     || ((const char*)(b) < (const char*)(a) && (const char*)(a) < ((const char*)(b) + (__fh_size_t)(len_b))) \
         )
 
 /*
